src/wasm-lib/grackle/src/lib.rs

```diff
@@ -387,6 +479,12 @@ pub enum CompileError {
         "you tried to read the '.{property}' of an object, but the object doesn't have any properties with that key"
     )]
     UndefinedProperty { property: String },
+    #[error("{0}")]
+    BadParamOrder(RequiredParamAfterOptionalParam),
+    #[error("A KCL function cannot have anything after its return value")]
```

Comment:
Wouldn't "can't have multiple return values" be more precise?

Reply:
Your suggested wording leaves out cases like this
```kcl
() => {
  return 1
  let x = 3
}
```

which I think mine covers